core/src/test/java/io/spine/change/DoubleMismatchTest.java

```diff
@@ -0,0 +1,150 @@
+
+    @Test
+    @DisplayName(HAVE_PARAMETERLESS_CTOR)
+    void haveUtilityConstructor() {
+        assertHasPrivateParameterlessCtor(DoubleMismatch.class);
+    }
+
+    @Test
+    @DisplayName(NOT_ACCEPT_NULLS)
+    void passNullToleranceCheck() {
+        new NullPointerTester()
+                .testAllPublicStaticMethods(DoubleMismatch.class);
+    }
+
+    @Nested
+    @DisplayName("create ValueMismatch instance")
+    class CreateMismatchTest {
+
+        @Test
+        @DisplayName("from given double values")
+        void createMismatchWithDoubles() {
+            final ValueMismatch mismatch = DoubleMismatch.of(EXPECTED, ACTUAL, NEW_VALUE, VERSION);
+
+            assertEquals(EXPECTED, unpackExpected(mismatch), DELTA);
+            assertEquals(ACTUAL, unpackActual(mismatch), DELTA);
+            assertEquals(NEW_VALUE, unpackNewValue(mismatch), DELTA);
+            assertEquals(VERSION, mismatch.getVersion());
+        }
+
+        @Test
+        @DisplayName("for expected zero amount")
+        void createForExpectedZero() {
+            final double expected = 0.0;
+            final ValueMismatch mismatch = expectedZero(ACTUAL, NEW_VALUE, VERSION);
+
+            assertEquals(expected, unpackExpected(mismatch), DELTA);
+            assertEquals(ACTUAL, unpackActual(mismatch), DELTA);
+            assertEquals(NEW_VALUE, unpackNewValue(mismatch), DELTA);
+            assertEquals(VERSION, mismatch.getVersion());
+        }
+
+        @Test
+        @DisplayName("for expected non zero amount")
+        void createForExpectedNonZero() {
+            final double actual = 0.0;
+            final ValueMismatch mismatch = expectedNonZero(EXPECTED, NEW_VALUE, VERSION);
+
+            assertEquals(EXPECTED, unpackExpected(mismatch), DELTA);
+            assertEquals(actual, unpackActual(mismatch), DELTA);
+            assertEquals(NEW_VALUE, unpackNewValue(mismatch), DELTA);
+            assertEquals(VERSION, mismatch.getVersion());
+        }
+
+        @SuppressWarnings("Duplicates") // Common test case for different Mismatches.
+        @Test
+        @DisplayName("for unexpected double value")
+        void createForUnexpectedDouble() {
+            final ValueMismatch mismatch = unexpectedValue(EXPECTED, ACTUAL, NEW_VALUE, VERSION);
+
+            assertEquals(EXPECTED, unpackExpected(mismatch), DELTA);
+            assertEquals(ACTUAL, unpackActual(mismatch), DELTA);
+            assertEquals(NEW_VALUE, unpackNewValue(mismatch), DELTA);
+            assertEquals(VERSION, mismatch.getVersion());
+        }
+    }
+
+    @Test
+    @DisplayName("not accept same expected and actual values")
+    void notAcceptSameExpectedAndActual() {
+        final double value = 19.19;
+        assertThrows(IllegalArgumentException.class,
+                     () -> unexpectedValue(value, value, NEW_VALUE, VERSION));
+    }
+
+    @Nested
+    @DisplayName("if given non-double ValueMismatch, fail to unpack")
+    class UnpackMismatchTest {
```

Comment:
Continuing the previous comment, this could be named `FailToUnpack`. And its methods are then going to be
* `expectedWithWrongType`;
* `actualWithWrongType`
* `newValueWithWrongType`.